float64 fromX         # x of starting location
float64 fromY         # y of starting location
float64 width         # width of rectangle
float64 height        # height of rectangle
